schlib/rules/rule3_12.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+
+from rules.rule import *
+
+class Rule(KLCRule):
+    """
+    Create the methods check and fix to use with the kicad lib files.
+    """
+    def __init__(self, component):
+        super(Rule, self).__init__(component, 'Rule 3.12 - Footprint filters', 'Footprint filters should match all appropriate footprints')
+
+    def checkFilters(self, filters):
+    
+        self.bad_filters = []
+   
+        for filter in filters:
+            errors = []
+            # Filter must contain a "*" wildcard
+            if not '*' in filter:
+                errors.append("Does not contain wildcard ('*') character")
+                
+            if len(filter.count(':')) > 1:
+                errors.append("Filter should not contain more than one (':') character")
+                
```

Comment:
What about my proposals:
```
                if re.search('(SOIC|SOIJ|SIP|DIP|SO|SOT-\d+|SOT\d+|QFN|DFN|QFP|SOP|TO-\d+|VSO|PGA|BGA|LLC|LGA)-\d+[W-_\*\?$]+', filter, flags=re.IGNORECASE) is not None:
                    self.warning("FPFilter {fcnt}: footprint filter '{filter}' seems to contain pin-number, but should not!".format(filter=filter, fcnt=fcnt))
                if ('-' in filter) or ('_' in filter):
                    self.warning("FPFilter {fcnt}: Minuses and underscores in footprint filter '{filter}' should be escaped with '?' or '*'.".format(filter=filter, fcnt=fcnt))

```

These are designed to help and note errors that are often done when people submit a symbol!